pkg/controller/daemon/daemon_controller.go

```diff
@@ -927,7 +927,19 @@ func (dsc *DaemonSetsController) syncNodes(ds *extensions.DaemonSet, podsToDelet
 		for i := pos; i < pos+batchSize; i++ {
 			go func(ix int) {
 				defer createWait.Done()
-				err := dsc.podControl.CreatePodsOnNode(nodesNeedingDaemonPods[ix], ds.Namespace, &template, ds, metav1.NewControllerRef(ds, controllerKind))
+
+				hostname := nodesNeedingDaemonPods[ix]
+				podTemplate := &template
+
+				if utilfeature.DefaultFeatureGate.Enabled(features.NoDaemonSetScheduler) {
+					// Clearup hostname, default scheduler will handle it.
+					hostname = ""
+					podTemplate = template.DeepCopy()
+					podTemplate.Spec.Affinity = util.AppendDaemonSetPodNodeAffinity(
```

Comment:
Please combine this and #59798 into one PR (ok to be separate commits)

Reply:
sure; let me combine them :)